lib/src/common/observer_controller.dart

```diff
@@ -501,34 +510,140 @@ mixin ObserverControllerForScroll on ObserverControllerForInfo {
     final targetChild = findCurrentFirstChild(obj);
     if (targetChild == null) return;
     final isHorizontal = obj.constraints.axis == Axis.horizontal;
-    var nextChild = obj.childAfter(targetChild);
-    nextChild ??= obj.childBefore(targetChild);
-    double separatorTotalHeight = 0;
-    if (nextChild != null && nextChild is! RenderIndexedSemantics) {
-      // It is separator
-      final nextChildPaintBounds = nextChild.paintBounds;
-      final nextChildSize = isHorizontal
-          ? nextChildPaintBounds.width
-          : nextChildPaintBounds.height;
-      separatorTotalHeight = index * nextChildSize;
-    }
     final childPaintBounds = targetChild.paintBounds;
-    final childSize =
+    double itemSeparatorHeight = 0;
+    int indexOfLine = index;
+    if (obj is RenderSliverList ||
+        obj is RenderSliverFixedExtentList ||
+        renderSliverType == ObserverRenderSliverType.list) {
+      // ListView
+      var nextChild = obj.childAfter(targetChild);
+      nextChild ??= obj.childBefore(targetChild);
+      if (nextChild != null && nextChild is! RenderIndexedSemantics) {
+        // It is separator
+        final nextChildPaintBounds = nextChild.paintBounds;
+        itemSeparatorHeight = isHorizontal
+            ? nextChildPaintBounds.width
+            : nextChildPaintBounds.height;
+      }
+    } else if (obj is RenderSliverGrid ||
+        renderSliverType == ObserverRenderSliverType.grid) {
```

Comment:
The processing logic of `_handleScrollToIndexForFixedHeight` method is too long, it is recommended to split out the corresponding two methods and return the data in a model.

Reply:
yes, it would be better